src/index.js

```diff
@@ -248,12 +247,16 @@ export default class InputNumber extends React.Component {
   }
 
   onChange = (e) => {
+    const {
+      onChange,
+    } = this.props;
+
     if (this.state.focused) {
       this.inputting = true;
     }
-    const input = this.props.parser(this.getValueFromEvent(e));
-    this.setState({ inputValue: input });
-    this.props.onChange(this.toNumberWhenUserInput(input)); // valid number or invalid string
+    this.rawInput = this.props.parser(this.getValueFromEvent(e));
+    this.setState({ inputValue: this.rawInput });
```

Comment:
Does `rawInput` need to be tracked on `this` context or can it be tracked in state via `inputValue`? Just wondering if we need to have `this.rawInput` at all

Reply:
Yes. I store it in `this` context because I use rawInput here: https://github.com/react-component/input-number/pull/156/files#diff-1fdf421c05c1140f6d71444ea2b27638R123 . If I use this.state.input value, then the dispaly text lags behind the actual value.